tests/graph/test_dot.py

```diff
@@ -128,3 +128,23 @@ def test_graph_no_args(self):
         graph = rustworkx.undirected_gnp_random_graph(3, 0.95, seed=24)
         dot_str = graph.to_dot()
         self.assertEqual("graph {\n0 ;\n1 ;\n2 ;\n2 -- 0 ;\n2 -- 1 ;\n}\n", dot_str)
+
```

Comment:
Can we add more tests, including:
* we go from PyGraph.to_dot and PyDiGraph.to_dot and back, covering multiple combinations of `node_attr`, `edge_attr` and `graph_attr`
*  A test case with an input string generated by https://networkx.org/documentation/stable/reference/generated/networkx.drawing.nx_pydot.write_dot.html